File: DSA/DynamicProgramming/Code/FibonacciDP.cs
// Fibonacci using DP (Tabulation) in C#

using System;

class FibonacciDP {
    static int Fib(int n) {
        int[] dp = new int[100];
        dp[0] = 0;
        dp[1] = 1;
        for (int i = 2; i <= n; i++) {
            dp[i] = dp[i-1] + dp[i-2];
        }
        return dp[n];
    }
    static void Main() {
        Console.WriteLine("=== Fibonacci using DP (C#) ===\n");
        int n = 10;
        Console.WriteLine($"Fibonacci({n}) = {Fib(n)}");
        Console.WriteLine("\nTime Complexity: O(n)");
        Console.WriteLine("Space Complexity: O(n)");
    }
}

File: DSA/DynamicProgramming/Code/FibonacciDP.py
# Fibonacci using DP (Tabulation) in Python

def fib(n):
    dp = [0] * (n+1)
    dp[0] = 0
    dp[1] = 1
    for i in range(2, n+1):
        dp[i] = dp[i-1] + dp[i-2]
    return dp[n]

n = 10
print("=== Fibonacci using DP (Python) ===\n")
print(f"Fibonacci({n}) = {fib(n)}")
print("\nTime Complexity: O(n)")
print("Space Complexity: O(n)")
